npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -164,27 +165,42 @@ def path_dependency_details(fetched_files)
         ].uniq
       end
 
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/MethodLength
       def path_dependency_details_from_manifest(file)
         return [] unless file.name.end_with?("package.json")
 
         current_dir = file.name.rpartition("/").first
         current_dir = nil if current_dir == ""
         path_dep_starts = %w(file: / ./ ../ ~/ link:.)
 
-        # Fetch yarn "file:" path "resolutions" so that we can resolve the
-        # lockfile. This pattern seems to be used to replace a sub-dependency
-        # with a local mock version.
-        dependency_types = NpmAndYarn::FileParser::DEPENDENCY_TYPES +
-                           ["resolutions"]
-        dependency_objects = JSON.parse(file.content).
-                             values_at(*dependency_types).
-                             compact
+        dep_types = NpmAndYarn::FileParser::DEPENDENCY_TYPES
+        parsed_manifest = JSON.parse(file.content)
+        dependency_objects = parsed_manifest.values_at(*dep_types).compact
+        # Fetch yarn "file:" path "resolutions" so the lockfile can be resolved
+        resolution_objects = parsed_manifest.values_at("resolutions").compact
+        manifest_objects = dependency_objects + resolution_objects
 
-        unless dependency_objects.all? { |o| o.is_a?(Hash) }
+        unless manifest_objects.all? { |o| o.is_a?(Hash) }
           raise Dependabot::DependencyFileNotParseable, file.path
         end
 
-        dependency_objects.flat_map(&:to_a).
+        # Re-write the glob name to the targeted dependency name (which is used
+        # in the lockfile), for example "parent-pacakge/**/sub-dep/target-dep" >
+        # "target-dep"
+        resolution_deps = resolution_objects.flat_map(&:to_a).map do |name, v|
+          # Picking the last two parts that might include a scope
+          parts = name.split("/").reverse.first(2)
```

Comment:
This read a bit confusing to me. Maybe replace with:

```
parts = name.split("/").reverse.last(2)
parts.pop if parts.count == 2 && !parts.first.start_with?("@"))
[parts.join("/"), v]
```

Reply:
(FYI I updated the above comment to make it actually work.)

Reply:
Maybe also put this in a function (`convert_dependency_path_to_name(path)`) to appease rubocop?

Reply:
Yes nice, cleaned this up and got rid of most of the disabled cops 😅